cmake/gsl_functions.cmake

```diff
@@ -5,68 +5,16 @@
 #
 # Any functions/macros should have a gsl_* prefix to avoid problems
 
-function(gsl_set_default_cxx_standard min_cxx_standard)
-    set(GSL_CXX_STANDARD "${min_cxx_standard}" CACHE STRING "Use c++ standard")
-
-    # when minimum version required is 3.8.0 remove if below
-    # both branches do exactly the same thing
-    if (CMAKE_VERSION VERSION_LESS 3.7.9)
-        if (MSVC)
-            set(GSL_CXX_STD_OPT "-std:c++${GSL_CXX_STANDARD}")
-        else()
-            set(GSL_CXX_STD_OPT "-std=c++${GSL_CXX_STANDARD}")
-        endif()
-
-        include(CheckCXXCompilerFlag)
-        CHECK_CXX_COMPILER_FLAG("${GSL_CXX_STD_OPT}" COMPILER_SUPPORTS_CXX_STANDARD)
-
-        if(COMPILER_SUPPORTS_CXX_STANDARD)
-            target_compile_options(GSL INTERFACE "${GSL_CXX_STD_OPT}")
-        else()
-            message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} has no c++${GSL_CXX_STANDARD} support. Please use a different C++ compiler.")
-        endif()
```

Comment:
This is the problematic code. It causes the following result in the `Microsoft.GSLConfig.cmake`:

```cmake
set_target_properties(Microsoft.GSL::GSL PROPERTIES
  INTERFACE_COMPILE_OPTIONS "-std=c++14" # <---------- This is problematic
)
```

This causes clients to receive the compiler flag. In the worst case scenario, clients have to filter or zero out the `INTERFACE_COMPILE_OPTIONS` property. Which isn't ideal.

Reply:
To be clear this is only a problem for clients using less than 3.8